### PR TITLE
[Frontend][Paddle]Handle Exception in Op Conversion.

### PR DESCRIPTION
**Pull request by zhangYiIntel**
### Details:
 - *Handle Exception in Op Conversion*
 - *Check Paddle Model Version when converting*

### Tickets:
 - *61122*

### CODE REVIEW

File: ngraph/frontend/paddlepaddle/src/frontend.cpp [hunk 1]
```diff
@@ -49,8 +49,15 @@ NamedOutputs make_ng_node(const std::map<pdpd::TensorName, Output<Node>>& nodes,
             named_inputs[input_port.parameter()].push_back(node_it->second);
         }
     }
+    NamedOutputs outputs;
+    // In case the conversion function throws exception
+    try {
+        outputs = creator_it->second(NodeContext(DecoderPDPDProto(op_place), named_inputs));
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/frontend.cpp:55]: Is it possible to create some tests for this?

File: ngraph/frontend/paddlepaddle/src/model.cpp [hunk 2]
```diff
@@ -289,7 +289,14 @@ InputModelPDPD::InputModelPDPDImpl::InputModelPDPDImpl(const std::basic_string<T
 
     FRONT_END_GENERAL_CHECK(pb_stream && pb_stream.is_open(), "Model file doesn't exist");
     FRONT_END_GENERAL_CHECK(m_fw_ptr->ParseFromIstream(&pb_stream), "Model can't be parsed");
-
+    // According to Paddle, the saved model has the framework version
+    // For exmaple Paddle 2.1.0 is encoded as 2001000. 0 means latest framework.
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/model.cpp:293]: typo: 'example'

File: ngraph/frontend/paddlepaddle/src/model.cpp [hunk 3]
```diff
@@ -289,7 +289,14 @@ InputModelPDPD::InputModelPDPDImpl::InputModelPDPDImpl(const std::basic_string<T
...
+    // https://github.com/PaddlePaddle/Paddle/blob/develop/cmake/version.cmake
+    // https://github.com/PaddlePaddle/Paddle/blob/2100816c5190693cc7dee181e96af72e9f0fbd1d/paddle/fluid/framework/program_desc.cc#L52
+    int64_t version = m_fw_ptr->version().version();
+    FRONT_END_GENERAL_CHECK(
+        version >= 2001000 || version == 0,
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/model.cpp:298]: I guess we can create test for this as well. As far as I remember @mvafin did some 'binary' modifications to simulate 'unsupported operation'. With same approach we could simulate 'unsupported version'

File: ngraph/frontend/paddlepaddle/src/frontend.cpp (hunk 1 above)
Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/frontend.cpp:55]: Yep, it needs to be tested. I think we can add an example that triggers an exception in op conversion

File: ngraph/frontend/paddlepaddle/src/model.cpp (hunk 3 above)
Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/model.cpp:298]: You are right, Let me figure out a way to simulate a lower version model file

File: ngraph/frontend/paddlepaddle/src/model.cpp (hunk 2 above)
Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/model.cpp:293]: sorry, will fix it

File: ngraph/frontend/paddlepaddle/src/model.cpp (hunk 3 above)
Comment (zhangYiIntel) [reply ngraph/frontend/paddlepaddle/src/model.cpp:298]: Tests enabled in 7a9e55f6a867a653d004e2dd8e86e0cb0bf27e78

**Comment by zhangYiIntel** [comment]
@nosovmik @ilyachur Could you help review ?

File: ngraph/frontend/paddlepaddle/src/model.cpp [hunk 4]
```diff
@@ -289,7 +289,14 @@ InputModelPDPD::InputModelPDPDImpl::InputModelPDPDImpl(const std::basic_string<T
 
     FRONT_END_GENERAL_CHECK(pb_stream && pb_stream.is_open(), "Model file doesn't exist");
     FRONT_END_GENERAL_CHECK(m_fw_ptr->ParseFromIstream(&pb_stream), "Model can't be parsed");
-
+    // According to Paddle, the saved model has the framework version
+    // For example Paddle 2.0.0 is encoded as 2001000. 0 means the latest framework.
```
Comment (nosovmik) [inline ngraph/frontend/paddlepaddle/src/model.cpp:293]: 2001000 corresponds to 2.1.0, not to 2.0.0

File: ngraph/test/frontend/paddlepaddle/throw_in_conversion.cpp [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ASSERT_NE(frontEnd, nullptr);
+    auto model_filename = FrontEndTestUtils::make_model_path(
+        std::string(TEST_PADDLE_MODELS_DIRNAME) + std::string("throw_in_conversion/throw_in_conversion.pdmodel"));
+    ASSERT_NO_THROW(inputModel = frontEnd->load(model_filename));
+    ASSERT_NE(inputModel, nullptr);
+    std::shared_ptr<ngraph::Function> function;
+    ASSERT_THROW(function = frontEnd->convert(inputModel), OpConversionFailure);
+}
+
+TEST(FrontEndConvertModelTest, unspported_version) {
```
Comment (nosovmik) [inline ngraph/test/frontend/paddlepaddle/throw_in_conversion.cpp:29]: typo: "unsupported"

**Review by nosovmik — APPROVED** [review]
Few very minor comments

**Comment by zhangYiIntel** [comment]
@ilyachur Could you have a review and merge the PR ?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**